### PR TITLE
 feat/simulate message

### PR DESCRIPTION
**Pull request by tynes**
## Description

Adds `simulateMessage` to the execution manager to be used with `eth_call`
Must be used with geth here: https://github.com/ethereum-optimism/go-ethereum/pull/210

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol [hunk 1]
```diff
@@ -1805,4 +1805,38 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
...
+        address _from,
+        iOVM_StateManager _ovmStateManager
+    )
+        external
+        returns (
+            bool,
+            bytes memory
+        )
+    {
+        // Prevent this call from having any effect unless at in a custom-set VM frame
```
Comment (tynes) [inline contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:1830]: Nit: `at in`, not sure if that is proper english

File: contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol [hunk 2]
```diff
@@ -1805,4 +1805,38 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         messageRecord.nuisanceGasLeft = 0;
         messageRecord.revertFlag = RevertFlag.DID_NOT_REVERT;
     }
+
+    /***********************
+     * L2-only Helper Functions *
+     ***********************/
```
Comment (ben-chain) [inline contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:1811]:
```suggestion
    /*****************************
     * L2-only Helper Functions *
     *****************************/
```
```suggestion
    /*********************
     * L2-only Helper Functions *
     *********************/
```

File: contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol (hunk 1 above)
Comment (ben-chain) [inline contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:1830]:
```suggestion
        // Prevent this call from having any effect unless in a custom-set VM frame
```

**Review by ben-chain — APPROVED** [review]
LGTM!

**Pull request merged**